src/testing/oem-debuggers.md

```diff
@@ -1,371 +1,493 @@
+     <div markdown="1">
+     Flutter app showing two buttons to open flutter.dev in a browser or within the app.
+     </div>
 
-* Choose the process you want to attach to. In this case, it's the
-  `com.google.clickcount`
-   (or <strong>com.<em>company</em>.<em>app_name</em></strong>)
-   process for the Motorola Moto G.
+1. Click the **Attach debugger to Android process** button.
+   (![Tiny green bug superimposed with a light grey arrow](/assets/images/docs/testing/debugging/oem/attach-process-button.png))
 
-  <img src='/assets/images/docs/testing/debugging/oem/choose-process-dialog.png' alt='screenshot containing two buttons for opening flutter.dev'>{:width="100%"}
+    {{site.alert.tip}}
+      If this button doesn't appear in the **Projects** menu bar, verify that
+      you opened Flutter _application_ project but _not a Flutter plugin_.
+    {{site.alert.end}}
 
-* In the debug pane, you should now see a tab for **Android Debugger**.
+1. The **process** dialog displays one entry for each connected device.
+   Select **show all processes** to display available processes for each
+   device.
 
-* In the project pane, expand
-  <nobr><strong><em>app_name</em> > android > app > src > main >
-  java > io.flutter plugins</strong></nobr>.
-  Double click **GeneratedProjectRegistrant** to open the
-  Java code in the edit pane.
+1. Choose the process to which you want to attach.
+   In following screenshot, the `com.example.my_app` process
+   using the **Emulator Pixel_5_API_33** was selected.
 
-Both the Dart and OEM debuggers are interacting with the same process.
-User either, or both, to set breakpoints, examine stack, resume execution...
-In other words, debug!
+   !['Flutter app in Android device displaying two buttons and a link.'](/assets/images/docs/testing/debugging/oem/choose-process-dialog.png){:width="50%"}
+   <div markdown="1">
+   Flutter app in Android device displaying two buttons and a link.
+   </div>
 
-  <img src='/assets/images/docs/testing/debugging/oem/dart-debugger.png' alt='screenshot of Android Studio in the Dart debug pane.'>{:width="100%"}
-  <br><center>The Dart debug pane with two breakpoints set in `lib/main.dart`</center>
+1. Locate the tab for **Android Debugger** in the **Debug** pane.
 
-  <img src='/assets/images/docs/testing/debugging/oem/android-debugger.png' alt='screenshot of Android Studio in the Android debug pane.'>{:width="100%"}
-  <br><center>The Android debug pane with one breakpoint set in
-   `GeneratedPluginRegistrant.java`.
-  Toggle between the debuggers by clicking the appropriate debugger in
-   the Debug pane's banner.</center>
+1. In the **Project** pane, expand
+   **_app_name__android > android > app > src > main > java > io.flutter plugins**.
 
-## Debugging with Xcode (iOS)
+1. Double click **GeneratedProjectRegistrant** to open the
```

Comment:
Why point them to the registration code rather than the actual plugin code? I can't imagine a scenario where this is the file a developer would be interested in debugging.

Reply:
Honestly? I don't know what file(s) to point to. The [original page](https://docs.flutter.dev/testing/oem-debuggers#debugging-with-android-gradle-android) pointed to the registration code.

Reply:
If the goal is to demonstrate debugging a plugin I would use the source of url_launcher_android, but I guess that makes things more fragile, so maybe that's not a good idea after all.